### ISSUE TITLE
non-image data

### ISSUE THREAD

**Issue by rojinsafavi** [junyanz/pytorch-CycleGAN-and-pix2pix#992]
I have two non-image modalities (samples are not paired). I know that the code needs to be changed since I'm working with non-image data,  but I was wondering if cycleGan principle and idea are applicable to non-image data? 

Thank you!

**Comment by junyanz** [comment]
It has been applied to non0image data (e.g., voice conversion [project](http://www.kecl.ntt.co.jp/people/kaneko.takuhiro/projects/cyclegan-vc/)). You probably need to implement your own generator and discriminator. The loss can still be used.

**Comment by rojinsafavi** [comment]
Thanks Junyan,

One more question, I'm going to explaining my dataset a bit more to get your insight into the application of the method.

I have 2 datasets, dataset 1 has n samples with 10 classes(labels). Dataset 2 has m samples with 5 classes. So I basically don't know what labels in dataset1 are similar to what samples in dataset2. Is CycleGAN still applicable in this scenario? Correct me if I am wrong, but it seems that you only trained the model on specific datasets each type ( for example dataset1 containing only horses and dataset2 only zebras, but not like dataset1 containing hourses+cats+dogs and dataset2 zebra+dog+fox), am I correct?

**Comment by junyanz** [comment]
We haven't experimented with this setting. Not sure if it will work. But it might be worth trying.

**Comment by guyshur** [comment]
Probably too late but have you seen https://arxiv.org/abs/1802.10151?
